### DISCUSSION TITLE
Copilot won't write new PHP code? Keeps writing C# instead?

### DISCUSSION THREAD

**Discussion by d-miles** [community/community#70111] [Copilot Conversations]
### Select Topic Area

Bug

### Body

Copilot chat can generate PHP, even while typing in a PHP file, Copilot will suggest new PHP code and it seems to work pretty well.

However, in VSCode if I open a new file, I'm asked to press `Cmd+I` to ask Copilot to do something:

<img width="671" alt="Screenshot 2023-10-14 at 12 54 39 PM" src="https://github.com/community/community/assets/16198745/9b0d917a-af5e-4f8c-acc3-b2d19a196f77">

When doing so, it just keeps generating C# instead of PHP: (in a 100% PHP project/workspace)

<img width="732" alt="Screenshot 2023-10-14 at 12 55 49 PM" src="https://github.com/community/community/assets/16198745/61d27deb-6fe1-4320-a755-3a19e04c1691">

I tried refreshing the result and being more specific (saying "my INamingStrategy interface" and specifically mentioning "in PHP") but got mixed results -- never once did it actually implement my interface though. One time, in the same file as the screenshot above, it basically just created a PHP version of the C# interface it had just generated? 

<img width="803" alt="Screenshot 2023-10-14 at 12 57 15 PM" src="https://github.com/community/community/assets/16198745/e1f2bd28-c57a-4ebd-9932-b964c57ee4aa">

So, I start manually implementing a class and interface (that I just made up as a test), and intellisense asks if I want to have Copilot fix the problem (class doesn't implement interface methods). VSCode presents a Copilot prompt with the following command:

```
/fix 'SnakeCaseNamingStrategy' does not implement methods 'getEntityAccessor', 'getTableName', 'getColumnName', 'getJoinTableName', 'getJoinColumnName'
```

When I send the command, it just says:

> Unrecognized language: php

See here:

<img width="948" alt="Screenshot 2023-10-14 at 12 42 06 PM" src="https://github.com/community/community/assets/16198745/ec75febc-32ca-4cb4-9f78-15e19822a09a">

Meanwhile, if I ask Copilot Chat the same thing, it works just fine. What gives?

<img width="965" alt="Screenshot 2023-10-14 at 1 02 47 PM" src="https://github.com/community/community/assets/16198745/82e396d2-db5c-44ed-9724-54302780a747">

Honestly, outside of the intellisense, I'm finding it difficult to find legitimate use cases for the chat and product, and the `Cmd+I` interface clearly doesn't work at all. Maybe this is just because it's not using GPT-4 (I'm signed up or on waiting list but Copilot X access might not be there yet?). There are times when it seems to get confused about MY project vs the chat context, also.

**Comment by lroush1** [comment]
+1  Also getting the "unrecognized language: php"

- Tested in VS Code (prod) and VS Code (insiders)
- Tested using CoPilot Prod and CoPilot Release Preview
Both of those configurations present the issue. I didn't used to have this problem before. I was doing great!

**Reply by tommmoe** [reply]
+1

glad my trial is ending soon.

**Comment by cryptolit0** [comment]
i am having the same issue but with svelte

**Comment by DPessovski** [comment]
+1

it is useless for PHP in comparison to js
